lib/src/deprecations.ts

```diff
@@ -25,53 +25,103 @@ export function getDeprecationIds(
   });
 }
 
+/**
+ * Map between active compilations and the deprecation options they use.
+ *
+ * This is used to determine which options to use when handling host-side
+ * deprecation warnings that aren't explicitly tied to a particular compilation.
+ */
+export const activeDeprecationOptions: Map<Object, DeprecationOptions> =
```

Comment:
If you're using `Symbol`s as the key here, you could actually just use a `Record<Symbol, DeprecationOptions>`. Either way, the key type should be `Symbol`.

Reply:
Changed the key type to Symbol, but keeping this as a Map because it's better for adding/removing and checking the size